### PR TITLE
format cache

### PR DESCRIPTION
**Pull request by elahehrashedi**

### CODE REVIEW

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: src/cache.ts [hunk 1]
```diff
@@ -249,14 +248,14 @@ export class CMakeCache {
 
                 // FIXME: How can `value` be boolean desipte being marked as string in the signature?
 
-                if (isString(value)) {
+                if (typeof value === 'string') {
```
[the remark this answers is not in the document]
Comment (elahehrashedi) [reply src/cache.ts:251]: it was deprecated

File: src/cache.ts [hunk 2]
```diff
@@ -249,14 +248,14 @@ export class CMakeCache {
...
                     const newlineIndex = value.search(/[\r\n]/);
                     if (newlineIndex >= 0) {
                         value = value.substring(0, newlineIndex);
                         log.warning(localize('cache.value.truncation.warning', 'Newline(s) found in cache entry {0}. Value has been truncated to {1}', `"${key}"`, `"${value}"`));
                     }
                 }
-                const newValueLine = entry.serializedKey + type + (isBoolean(value) ? (value ? "TRUE" : "FALSE") : value);
+                const newValueLine = entry.serializedKey + type + ((typeof value === 'boolean') ? (value ? "TRUE" : "FALSE") : value);
```
[the remark this answers is not in the document]
Comment (elahehrashedi) [reply src/cache.ts:258]: it was deprecated, this was recommended instead

File: src/cache.ts [hunk 3]
```diff
@@ -103,33 +102,33 @@ export class CMakeCache {
...
-        private readonly _entries: Map<string, Entry>) {}
+    private constructor(private readonly cachePath: string,
+        private readonly cacheFileExists: boolean,
+        private readonly cacheEntries: Map<string, Entry>) {}
 
     /**
      * `true` if the file exists when this instance was created.
      * `false` otherwise.
      */
     get exists() {
```
[the remark this answers is not in the document]
Comment (elahehrashedi) [reply src/cache.ts:123]: `cacheFileExists ` was never used, removed it totally.

File: src/cache.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bobbrow) [reply src/cache.ts:217]: We have these functions in `util.ts` and they are not deprecated.  I thought it was using those.  Can you update it to use our functions?

File: src/cache.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (bobbrow) [reply src/cache.ts:224]: We have these functions in `util.ts` and they are not deprecated.  I thought it was using those.  Can you update it to use our functions?

**Review by bobbrow — APPROVED** [review]

**Review by bobbrow — APPROVED** [review]

**Pull request merged**